third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -575,7 +575,11 @@ struct AtomicRMWOpConversion
     // tensor
     if (tensorTy) {
       auto valTy = cast<RankedTensorType>(val.getType());
-      vec = std::min<unsigned>(vec, valTy.getElementType().isF16() ? 2 : 1);
+      Type elTy = valTy.getElementType();
+      vec = std::min<unsigned>(vec, llvm::isa<FloatType>(elTy) &&
```

Comment:
I think we should consider another case that elemsPerThreads = 1, then the vec size should be 1 instead of 2 since each threads has only element, no need to vectorized input. Is my understanding correct?

Reply:
it was already considered in ``vec`` initialization here: https://github.com/triton-lang/triton/pull/4925/files#diff-ad0e4d897f572806e62894967ba7b38db995b823a8b15dcd7f6eca0b1c89f667R573

this line allows only 16 bit float types to be vectorized.